src/mappostgis.cpp

```diff
@@ -1911,121 +1911,130 @@ static std::string msPostGISBuildSQLWhere(layerObj *layer, const rectObj *rect,
     return std::string();
   }
 
+  const rectObj rectInvalid = MS_INIT_INVALID_RECT;
+  bool bIsValidRect = memcmp(&rect, &rectInvalid, sizeof(rect)) != 0;
```

Comment:
you are comparing the address of the pointer rect, instead of the pointed value, so:
```suggestion
  bool bIsValidRect = memcmp(rect, &rectInvalid, sizeof(rectInvalid)) != 0;
```

Reply:
Thanks so much @rouault ! Changing that line fixed the debug output of the line above before reaching it..